### PR TITLE
Swagger for search

### PR DESCRIPTION
**Pull request by karannaoh**

### CODE REVIEW

**Comment by karannaoh** [comment]
#630

**Comment by karannaoh** [comment]
#616

File: guillotina/api/service.py [hunk 1]
```diff
@@ -45,11 +45,40 @@ class Service(View):
     __validator__ = __schema__ = None
     _sentinal = object()
 
+    def _validate_parameters(self):
+        if 'parameters' in self.__config__:
+            data = self.request.url.query
+            for parameter in self.__config__['parameters']:
+                if parameter['in'] == 'query':
+                    if parameter['schema']['type'] == 'integer':
```
Comment (vangheem) [inline guillotina/api/service.py]: Could we handle KeyError here?

File: guillotina/api/service.py [hunk 2]
```diff
@@ -45,11 +45,40 @@ class Service(View):
...
+                        try:
+                            int(data[parameter['name']])
```
Comment (vangheem) [inline guillotina/api/service.py]: Before all this, shouldn't we first check if `name` is in `data`?

File: guillotina/api/service.py [hunk 3]
```diff
@@ -45,11 +45,40 @@ class Service(View):
...
+                        try:
+                            float(data[parameter['name']])
+                        except ValueError:
+                            raise HTTPPreconditionFailed(content={
+                                'reason': 'Schema validation error',
+                                'message': 'can not convert {} to Float'.format(data[parameter['name']])
+                            })
+                    else:
+                        pass
+                    if parameter['required'] and parameter['name'] not in data:
```
Comment (vangheem) [inline guillotina/api/service.py]: what if they don't define `required`?

Maybe the whole check should be wrapped in `KeyError` handling?

File: guillotina/api/service.py (hunk 1 above)
Comment (karannaoh) [reply guillotina/api/service.py]: keyError for parameter['schema']['type']?

Comment (vangheem) [reply guillotina/api/service.py]: Yes, if they define a parameter without the `schema` defined

I imagine it already default to string so you don't need to define this.

**Pull request merged**